packages/http-server/src/server.ts

```diff
@@ -10,63 +11,75 @@ import { IPrismDiagnostic } from '@stoplight/prism-core';
 import { pipe } from 'fp-ts/lib/pipeable';
 import * as TE from 'fp-ts/lib/TaskEither';
 import * as E from 'fp-ts/lib/Either';
+import { MicriHandler } from 'micri';
+
+function searchParamsToNameValues(searchParams: URLSearchParams): IHttpNameValues {
+  const params = {};
+  for (const key of searchParams.keys()) {
+    const values = searchParams.getAll(key);
+    params[key] = values.length === 1 ? values[0] : values;
+  }
+  return params;
+}
+
+function addressInfoToString(address: AddressInfo | string | null) {
+  if (!address) return '';
+  const a = address as AddressInfo;
+  return `http://${a.address}:${a.port}`;
+}
+
+function parseRequestBody(request: IncomingMessage) {
+  // if no body provided then return null instead of empty string
+  if (
+    request.headers['content-type'] === undefined
+    && request.headers['transfer-encoding'] === undefined
+    && (request.headers['content-length'] === '0' || request.headers['content-length'] === undefined)
+  ) {
+    return Promise.resolve(null);
+  }
+
+  if (typeIs(request, ['application/json', 'application/*+json'])) {
+    return json(request);
+  } else {
+    return text(request);
+  }
+}
 
 export const createServer = (operations: IHttpOperation[], opts: IPrismHttpServerOpts): IPrismHttpServer => {
   const { components, config } = opts;
 
-  const server = fastify({
-    logger: components.logger,
-    disableRequestLogging: true,
-    modifyCoreObjects: false,
-  });
-
-  if (opts.cors) server.register(fastifyCors, { origin: true, credentials: true });
-
-  server.addContentTypeParser('*', { parseAs: 'string' }, (req, body, done) => {
-    if (typeIs(req, ['application/*+json'])) {
-      try {
-        return done(null, JSON.parse(body));
-      } catch (e) {
-        return done(e);
-      }
-    }
+  const handler: MicriHandler = async (request, reply) => {
+    const {
+      url,
+      method,
+      headers,
+      socket,
+    } = request;
 
-    if (typeIs(req, ['application/x-www-form-urlencoded'])) {
-      return done(null, body);
+    if (!url) {
+      return TE.left(new Error('Missing URL in request!'));
     }
 
-    const error: Error & { status?: number } = new Error(`Unsupported media type.`);
-    error.status = 415;
-    Error.captureStackTrace(error);
-    return done(error);
-  });
-
-  const prism = createInstance(config, components);
+    const body = await parseRequestBody(request);
 
-  const replyHandler: fastify.RequestHandler = (request, reply) => {
-    const {
-      req: { method, url },
-      body,
-      headers,
-      query,
-    } = request;
+    const { searchParams, pathname } = new URL(url, addressInfoToString(socket.address()));
```

Comment:
Given all we want to do here is separate the searchParams and the pathname, we can use a fake url so we do not have to call `socket.address` for each request

```suggestion
    const { searchParams, pathname } = new URL(url, 'http://IDoNotCare');
```

Although I'm wondering if there's an alternative…

Reply:
The alternative might be `urijs`. But it adds extra dependency vs. built-in URL&URLSearchParams.

I changed to `http://example.com` however, I feel like replacing magic with sorcery ;)